tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/AIBotConfig.java

```diff
@@ -57,18 +56,19 @@ public AIBotConfig(String apiKey, MailAddress botAddress, LlmModel llmModel, Opt
         this.llmModel = llmModel;
     }
 
-    public static AIBotConfig fromMailetConfig(MailetConfig mailetConfig) throws MailetException {
-        String apiKeyParam = mailetConfig.getInitParameter(API_KEY_PARAMETER_NAME);
-        String gptAddressParam = mailetConfig.getInitParameter(BOT_ADDRESS_PARAMETER_NAME);
-        String llmModelParam = mailetConfig.getInitParameter(MODEL_PARAMETER_NAME);
-        String baseUrlParam = mailetConfig.getInitParameter(BASE_URL_PARAMETER_NAME);
+    public static AIBotConfig from(Configuration configuration) throws IllegalArgumentException {
+        String apiKeyParam = configuration.getString(API_KEY_PARAMETER_NAME, "");
+        String gptAddressParam = configuration.getString(BOT_ADDRESS_PARAMETER_NAME, "");
+        String llmModelParam = configuration.getString(MODEL_PARAMETER_NAME, "");
+        String baseUrlParam = configuration.getString(BASE_URL_PARAMETER_NAME, "");
```

Comment:
Is `""` a sane usable default?

If we cannot figure out good defaults then lets reject missing values here

`Optional.ofNullable(configuration.getString(API_KEY_PARAMETER_NAME, null)).orElseThrow(() -> new IllegalArgumentException("...")));`